code/modules/reagents/Chemistry-Reagents.dm

```diff
@@ -51,6 +53,15 @@
 	return
 
 /datum/reagent/proc/on_mob_life(mob/living/carbon/M, alien)
+	if(length(purge_list))
```

Comment:
Get `remove_all_type()` to work on with list of reagents if you can, so we won't need this.
Otherwise move it to a holder proc, with purge_list and rate as args. This is not the place.

Reply:
Please do move this to the holder file if you think this is pertinent to the PR, thanks.